cmd/state-offline-installer/runInstall.go

```diff
@@ -0,0 +1,187 @@
+const assetsPathName = "assets"
+const artifactsPathName = "artifacts"
+const licenseFileName = "LICENSE.txt"
+
+func runInstall(out output.Outputer, params *Params) error {
+
+	analytics := blackhole.New()
+	prompter := prompt.New(true, analytics)
+	default_boolean_answer := true
+
+	tempDir, err := ioutil.TempDir("", "artifacts-")
+	if err != nil {
+		return errs.Wrap(err, "Unable to create temporary directory")
+	}
+
+	defer os.RemoveAll(tempDir)
+
+	out.Print(fmt.Sprintf("Temp directory is: %s", tempDir))
+
+	installToDir := params.path
+	assetsPath := filepath.Join(tempDir, assetsPathName)
+	artifactsPath := filepath.Join(tempDir, artifactsPathName)
+	licenseFilePath := filepath.Join(assetsPath, licenseFileName)
+
+	// Double check if installation directory exists already
+	if fileutils.DirExists(installToDir) {
+		empty, err := fileutils.IsEmptyDir(installToDir)
+		if err != nil {
+			return errs.Wrap(err, "Test for directory empty failed")
+		}
+		if !empty {
+			installNonEmpty, err := prompter.Confirm("Setup", "Installation directory is not empty, install anyway?", &default_boolean_answer)
+			if err != nil {
+				return errs.Wrap(err, "Unable to get confirmation to install into non-empty directory")
+			}
+
+			if !installNonEmpty {
+				return locale.NewInputError("offline_installer_err_installdir_notempty", "Installation directory ({{.V0}}) not empty, installation aborted", installToDir)
+			}
+		}
+	}
+
+	if err := os.Mkdir(assetsPath, os.ModePerm); err != nil {
+		return errs.Wrap(err, "Unable to create assetsPath")
+	}
+
+	if err := os.Mkdir(artifactsPath, os.ModePerm); err != nil {
+		return errs.Wrap(err, "Unable to create artifactsPath directory")
+	}
+
+	err = unzip(params.backpackZipFile, assetsPath)
+	if err != nil {
+		return errs.Wrap(err, "Unable to unzip to assetsPath")
+	}
+
+	eulaAccepted, err := showEULAAndGetAcceptance(&prompter, licenseFilePath)
+	if err != nil {
+		return errs.Wrap(err, "Error with EULA acceptance")
+	}
+
+	if !eulaAccepted {
+		return locale.NewInputError("err_eula_not_accepted", "End user didn't accept the EULA")
+	}
+
+	out.Print(fmt.Sprintf("Stage 1 of 2 Start: Decompressing assets into: %s", assetsPath))
+	err = unzip(params.backpackZipFile, assetsPath)
+	if err != nil {
+		return errs.Wrap(err, "Error unzipping assets")
+	}
+	out.Print(fmt.Sprintf("Stage 1 of 2 Finished: Decompressing assets into: %s", assetsPath))
+
+	archivePath := filepath.Join(assetsPath, artifactsTarGZName)
+
+	out.Print(fmt.Sprintf("Stage 2 of 2 Start: Decompressing artifacts into: %s", artifactsPath))
+	err = unpackArtifact(unarchiver.NewTarGz(), archivePath, artifactsPath)
+	if err != nil {
+		return errs.Wrap(err, "Could not unpack artifact %s", archivePath)
+	}
+	out.Print(fmt.Sprintf("Stage 2 of 2 Finished: Decompressing artifacts into: %s", artifactsPath))
+
+	offlineTarget := target.NewOfflineTarget(installToDir, artifactsPath)
+
+	// FIX: Need to do this more correctly with a New function
```

Comment:
We have a rule on our repo which is that we are not allowed to put todo's in code comments as our code is not a task tracker. I'd say either address this, remove it, or file a jira ticket to follow up at a later date.